io/src/main/scala/fs2/io/tcp/Socket.scala

```diff
@@ -236,28 +236,34 @@ protected[tcp] object Socket {
         // buffer is also reset to be ready to be written into.
         def getBufferOf(sz: Int): F[ByteBuffer] =
           bufferRef.get.flatMap { buff =>
-            if (buff.capacity() < sz)
+            if (buff.capacity() < sz) {
               bufferRef
                 .modify { _ =>
```

Comment:
is this any different from `setSync(ByteBuffer.allocatte(sz)`?

Reply:
We'd need `setAndGet` or otherwise have to write `F.delay(ByteBuffer.allocate).flatMap(b => ref.setSync(b).as(b))`

Reply:
sorry, edited too late. `flatTap` should make that more bearable, but again, not a big deal at all

Reply:
Yay for `flatTap` -- changed!